File: 96A/96A.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){
    string s;
    int duplicate = 0;
    bool flag = false;

    cin >> s;

    for(int i = 1; i < s.size(); i++){
        if(s[i - 1] == s[i]){
            duplicate ++;
        }
        else{
            duplicate = 0;
        }

        if(duplicate == 6){
            flag = true;
            break;
        }
    }

    if(flag == true){
        cout << "YES" << endl;
    }
    else{
        cout << "NO" << endl;
    }
}

File: 900/96A/96A.py
s = list(input())
duplicate = 0
flag = False

for i in range(1, len(s)):
    if s[i - 1] == s[i]:
        duplicate += 1
    else:
        duplicate = 0
        
    if duplicate == 6:
        flag = True
        break

if flag == True:
    print("YES")
else:
    print("NO")